src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -407,6 +407,9 @@ public static void main(String args[]){
       if(args[0].equals("runByCategory")){
         boolean validCategoryName = false;
         for(ModuleID category : ModuleID.values()){
+          if(category!="UI"){
+            category = category.toLowerCase());
+          }
```

Comment:
This can throw an error. category is of type ModuleID. category.name() will give the string but you cannot store it back in category